src/platform/mt793x/DnssdImpl.cpp

```diff
@@ -335,9 +335,14 @@ CHIP_ERROR ChipDnssdBrowse(const char * type, DnssdServiceProtocol protocol, chi
 
     (void) addressType;
     ChipLogProgress(ServiceProvisioning, "ChipDnssdBrowse %s", StringOrNullMarker(type));
-    strcpy(ServiceType, type);
-    strcat(ServiceType, ".");
-    strcat(ServiceType, GetProtocolString(protocol));
+    int ret = snprintf(ServiceType, sizeof(ServiceType), "%s.%s", type, GetProtocolString(protocol));
```

Comment:
consider using https://github.com/project-chip/connectedhomeip/blob/master/src/lib/support/StringBuilder.h#L28 instead of snprintf and return values.

Then you can use `Fit()` for clearer code than negative return values or casts of size for snprintf debug.